### PR TITLE
Fix lockup when no midi devices present.

### PR DESCRIPTION
**Pull request by 1st-BrainStormer**
# Test Plan
## Tools Required:
- loopMidi [https://www.tobias-erichsen.de/software/loopmidi.html](url) (midi loopback)
- Midi Keyboard/Controller or VMPK http://vmpk.sourceforge.net/(url)
- midiTest.js [https://brainstormer.keybase.pub/High%20Fidelity/Project/midi/midiTest.js](url)
### Install VMPK (IF you don't have a Keyboard/Midi Controller)
- Edit->Midi Connections
    - Turn off Midi input
    - Select Windows MM MIDI Out Driver
    - Select loopMidiPort Output Midi Connection
### Run midiTest.js
D#Device, CH#Channel, T:Type, N:Note/Controller V:Velocity/Pressure B:Pitch Bend P:Program
- Every 5 seconds the test will output 7 tests verifying the varius supported message types.
- With a keyboard/controller or VMPK you can also test the various signals:
    -  Note Off (Some instuments use note on with velocity 0 instead)
    -  Note On
    -  Polyphonic Key Pressure  (if you have keyboard/Midi Controller that supports this)
    -  Control Change
    -  Program Change
    -  Channel Pressure (if you have keyboard/Midi Controller that supports this)
    -  Pitch Bend Chamge (Note: VMPK doesn't send correctly)
    -  Change midi channel
  
Test running interface with no midi devices plugged in, verify that interface does not crash.
Rotate avatar, replug or power cycle midi device. There should be no pauses

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6465/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6465/HighFidelity-Beta-PR12214-bb09c71bda983d59cf5073b6ef31614a831369bf.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6465/HighFidelity-Beta-PR12214-bb09c71bda983d59cf5073b6ef31614a831369bf.exe

**Comment by themelissabrown** [comment]
@1st-BrainStormer can you add a test plan to this?

**Comment by MiladNazeri** [comment]
@1st-BrainStormer Looks like there is an issue with this PR.  Midi.cpp/Midi.h appear to be on the root?  
I think also you might need to check that you are using spaces instead of tabs.  I think I've seen mixed tabs and spaces on another file you sent me and wondering if that is why it's having a hard time parsing for a diff.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6475/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6476/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6477/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6477/HighFidelity-Beta-PR12214-e3fb9cee00c9dc3883bef05e19687499582e70b0.dmg

File: libraries/midi/src/Midi.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@ const int MIDI_BYTE_MASK = 0x0FF;
 const int MIDI_NIBBLE_MASK = 0x00F;
 const int MIDI_PITCH_BEND_MASK = 0x3F80;
 const int MIDI_SHIFT_STATUS = 4;
-const int MIDI_SHIFT_NOTE = 8;
+const int MIDI_SHIFT_NOTE = 8;-
```
Comment (MiladNazeri) [inline libraries/midi/src/Midi.cpp]: Looks like there is an extra - here

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6483/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6483/HighFidelity-Beta-PR12214-fab13a111c8a15213597712aabe110463b8bbca4.dmg

File: libraries/midi/src/Midi.cpp [hunk 2]
```diff
@@ -174,7 +174,7 @@ void Midi::sendMessage(int device, int channel, int type, int note, int velocity
 }
 
 void Midi::sendNote(int status, int note, int velocity) {
-    for (int i = 0; i < midihout.size(); i++) {
+    for (int i = 1; i < midihout.size(); i++) {  //  Skip 0 (Microsoft GS Wavetable Synth)
         if (midihout[i] != NULL) {
             midiOutShortMsg(midihout[i], status + (note << MIDI_SHIFT_NOTE) + (velocity << MIDI_SHIFT_VELOCITY));
```
Comment (kencooke) [inline libraries/midi/src/Midi.cpp]: small nitpick, for consistency this should be
```status | (note << MIDI_SHIFT_NOTE) | (velocity << MIDI_SHIFT_VELOCITY)); ```

**Comment by MiladNazeri** [comment]
build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6492/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6492/HighFidelity-Beta-PR12214-fab13a111c8a15213597712aabe110463b8bbca4.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6492/HighFidelity-Beta-PR12214-fab13a111c8a15213597712aabe110463b8bbca4.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6502/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6502/HighFidelity-Beta-PR12214-cbc8698e9a3273f12c7ccdbc79b089eb2bbecd15.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6502/HighFidelity-Beta-PR12214-cbc8698e9a3273f12c7ccdbc79b089eb2bbecd15.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/6511/app_12214_PR.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/6511/HighFidelity-Beta-PR12214-f7caba5295680e713c45849a7b97654ac3123281.exe

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/6511/HighFidelity-Beta-PR12214-f7caba5295680e713c45849a7b97654ac3123281.dmg

**Comment by sbennett77** [comment]
Currently Testing.

**Comment by sbennett77** [comment]
Passed.

@conklin94122

**Pull request merged**